### ISSUE TITLE
missing <pre> in \include breaks code indentation in w3m/lynx (Origin: bugzilla #680073)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4806]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.8.1.2* on platform *Other*
Assigned to: Dimitri van Heesch

On 2012-07-17 07:11:38 +0000, Brice Goglin wrote:

> Hello,
> 
> In earlier doxygen versions (at least in 1.7.6.1), including a C file with \include will produce some code between:
> <div class="fragment"><pre class="fragment">
> </pre></div>
> 
> Now we get:
> <div class="fragment">
> </div>
> with <div class="line"></div> around each line (from what I understand).
> 
> This causes some problems with browsers such as w3m and lynx because the code is not indented anymore. See for instance:
>   http://www.open-mpi.org/projects/hwloc/doc/v1.5rc1/#interface_example
> 
> By the way, maybe related, I can't find where div.line is defined in the CSS.
> 
> Is there any way to work around this problem? Otherwise I will have to patch the doxygen HTML output :/
> 
> thanks
> Brice

On 2012-07-17 07:28:23 +0000, Dimitri van Heesch wrote:

> The div.line is present in doxygen.css produced by doxygen (maybe you replaced it with your own, in which case you need to update your CSS). The CSS definition specifies that spaces have to be preserved. I used it for better markup possibilities and to allow wrapping of long lines.

On 2012-07-17 07:49:56 +0000, Brice Goglin wrote:

> Thanks I found the div.line stuff. But that doesn't seem to work with any text-based browser unfortunately. I tried using the vanilla doxygen.css, same problem. Nothing is indented. Given all the browser-specific-hacks in the CSS, I wonder if there's a way to make this work.

**Comment by bgoglin** [comment]
I'll give a look to confirm whether it's fixed.

**Comment by bgoglin** [comment]
The problem still exists with doxygen 1.9.1 which was used to generate https://www-lb.open-mpi.org/projects/hwloc/doc/v2.9.0/a00358.php#interface_example
The CSS is identical to what doxygen 1.9.1 generates.
If you view that page with w3m, lynx or links, you see that functions "print_children" and "main" are not indented a all.

**Comment by albert-github** [comment]
I think the essential point here is:
> If you view that page with w3m, lynx or links, you see that functions "print_children" and "main" are not indented at all.

as far as I can tel these 3 web browsers are so called text based web browser and I don't think they have not been considered.

@bgoglin do you have any idea what should be changed to be able to support also these browsers?

@doxygen what do we do with these type of browsers?


)(Note the current doxygen version is 1.9.6)

**Comment by bgoglin** [comment]
Yes this bug is only about these text browsers as specified in the title :)

"<pre>" was working very well unfortunately, I don't know how to make <div> generate code that these browsers will understand as similar to pre.

For the record, we use this to export the doxygen output to pure text format (basically for creating a README). If there's a way to do that directly in doxygen, I'll be happy to use it and stop having w3m/lynx/links do the conversion from HTML to text.

**Comment by albert-github** [comment]
Is it just for creating the README or also for other pure text formatted files?
Where can I find an example of the original HTML and converted text README as well as the original source. On GitHub I see a README (https://github.com/open-mpi/hwloc/blob/master/README) is this the desired result or an input?

**Comment by bgoglin** [comment]
Yes that README is an example of output. We generate the doc with doxygen, and then call w3m or lynx on a HTML page to generate that README. The source of that one is https://www-lb.open-mpi.org/projects/hwloc/doc/v2.9.0/). That one doesn't show any issue because we basically started truncating the output of w3m/lynx to avoid showing sections with indentation errors.

Before truncating, the output would be much longer, like https://github.com/open-mpi/hwloc/blob/hwloc-1.5.2/README generated from https://www-lb.open-mpi.org/projects/hwloc/doc/v1.5.2/
I don't remember if we manually reindented the output or not. It's been a while.
